db/migration/1665135955821-AddRedirectsTablesSQLFile.sql

```diff
@@ -0,0 +1,1773 @@
+-- This file is the SQL only content of the TypeORM migration of the same name.
+-- It exists for review purposes etc and is not intended to be run directly (although
+-- you can of course run fragments of this file against our MySQL database if you like)
+
+-- ----------------------------------------------
+-- TABLES
+-- table for manual bulk redirects, for example /app/uploads/* -> /uploads/:splat
+-- these are not fully taken into account as the wildcards are not interpreted
+-- but stored here for completeness
+CREATE TABLE
+  manual_bulk_redirects (
+    id int auto_increment primary key,
+    -- source slug, i.e. what should be matched by this redirect
+    -- will be sanizited by triggers to never include a trailing slash unless it is the root path
+    slug varchar(2048) not null,
+    -- target domain to link to, can be empty which is understood to mean ourworldindata.org
+    targetDomain varchar(255) not null default '',
+    -- target path, e.g. /some/path
+    -- will be sanizited by triggers to never include a trailing slash unless it is the root path
+    targetPath varchar(2048) not null,
+    -- target query (e.g. ?somekey=somevalue )
+    targetQuery varchar(2048) not null default '',
+    -- target fragment (e.g. #somefragment )
+    targetFragment varchar(255) not null  default '',
+    -- which kind of redirect to do, can be 301 or 302
+    statusCode int not null,
+    -- generated column with the domain and path concatted. This excludes query and fragment and is useful
+    -- if you e.g. want to know which sites link to the same path, regardless of query and/or fragment
+    targetLocation varchar(2100) generated always as (concat(targetDomain, targetPath)) virtual not null,
+    -- generated column with the full target url, including query and fragment
+    targetUrl varchar(5000) generated always as (
+      concat(
+        targetDomain,
+        targetPath,
+        targetQuery,
+        targetFragment
+      )
+    ) virtual not null,
+    -- md5 hashes. MySQL doesn't like check constraints on long VARCHARs so we use hashes to ensure
+    -- some uniqueness constraints like source <> target or more complex ones that avoid redirect chains
+    slugMd5 char(32) generated always as (md5(slug)) stored unique not null,
+    targetUrlMd5 char(32) generated always as (md5(targetUrl)) stored not null,
+    targetLocationMd5 char(32) generated always as (md5(targetLocation)) stored not null,
+    constraint manual_bulk_redirects_slug_target_must_be_different check (slugMd5 <> targetUrlMd5)
```

Comment:
```suggestion
    constraint manual_bulk_redirects_slug_target_must_be_different check (slugMd5 != targetUrlMd5)
```

nit: I like the `!=` syntax way more than the `<>`.